test.c

```diff
@@ -955,17 +955,28 @@ _indexer_writes (const char *d)
     return NULL;
 }
 
+static char*
+_provide_callback (const char *path)
+{
+    return strdup ("up");
+}
+
 void
 test_index_writes ()
 {
     char *path = TEST_PATH"/counters/*";
+    char *provide_path = TEST_PATH"/counters/*/something/else";
     GNode *root;
 
     CU_ASSERT (apteryx_index (path, _indexer_writes));
-    CU_ASSERT ((root = apteryx_get_tree (TEST_PATH"/counters")) == NULL);
+    /* There's no reason to call the indexer without a provider behind it. */
```

Comment:
Do we have a test for a leaflist e.g. TEST_PATH"/users" where it is a list of usernames only - in other words there is no provider and the indexer is providing all the data?